### ISSUE TITLE
ppc64le support added in BenchmarkDotNet is not picked in 'performance' repo

### ISSUE THREAD

**Issue by anjiltech** [dotnet/performance#2787]
The changes delivered with dotnet/BenchmarkDotNet#2216 are not reflected in the https://github.com/dotnet/performance repo. 

While running the microbenchmarks from within 'performance' repo, it fails as below:

// Validating benchmarks:

Unhandled Exception:
System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values.
at BenchmarkDotNet.Portability.RuntimeInformation.GetCurrentPlatform()
at BenchmarkDotNet.Portability.RuntimeInformation.GetArchitecture()
at BenchmarkDotNet.Environments.BenchmarkEnvironmentInfo..ctor()
at BenchmarkDotNet.Environments.HostEnvironmentInfo..ctor()
at BenchmarkDotNet.Environments.HostEnvironmentInfo.GetCurrent()
at BenchmarkDotNet.Toolchains.Toolchain.Validate(BenchmarkCase benchmarkCase, IResolver resolver)+MoveNext()
at BenchmarkDotNet.Toolchains.Roslyn.RoslynToolchain.Validate(BenchmarkCase benchmarkCase, IResolver resolver)+MoveNext()
at System.Collections.Generic.LargeArrayBuilder1[[BenchmarkDotNet.Validators.ValidationError, BenchmarkDotNet, Version=0.13.2.2009, Culture=neutral, PublicKeyToken=aa0ca2f9092cefc4]].AddRange(IEnumerable1 items)
at System.Collections.Generic.EnumerableHelpers.ToArray[ValidationError](IEnumerable1 source) at System.Linq.Enumerable.ToArray[ValidationError](IEnumerable1 source)
at BenchmarkDotNet.Running.BenchmarkRunnerClean.<>c__DisplayClass12_0.b__2(BenchmarkCase benchmark)
at System.Linq.Enumerable.WhereArrayIterator1[[BenchmarkDotNet.Running.BenchmarkCase, BenchmarkDotNet, Version=0.13.2.2009, Culture=neutral, PublicKeyToken=aa0ca2f9092cefc4]].ToArray() at System.Linq.Enumerable.ToArray[BenchmarkCase](IEnumerable1 source)

Also observed that the BenchmarkDotnet.dll is not built for net7.0 within BenchmarkDotnet repository.

**Comment by cincuranet** [comment]
This duplicates #2778. I'll close that one and leave this one.

**Comment by cincuranet** [comment]
@adamsitnik Branch `main` is [using](https://github.com/dotnet/performance/blob/main/eng/Versions.props#L18) `0.13.2.2009`. Is there a newer version (I don't see newer version in obvious places)?

**Comment by adamsitnik** [comment]
@cincuranet BDN publishes a new NuGet package for every green build of its master branch. But it pushes it to BDNs feed: https://ci.appveyor.com/nuget/benchmarkdotnet. dotnet/peformance has its own mirror feed, maintained internally by the .NET Performance Team. 

So to tell the long story short someone from Perf Team who has the access to the internal feed needs to upload a newer version to the mirror feed.

From BDN perspective the list of builds can be found at https://ci.appveyor.com/project/dotnetfoundation/benchmarkdotnet/history When you select given build you can go to artifacts tab to see the packages that were created:

![image](https://user-images.githubusercontent.com/6011991/207019166-34c2ddf2-b2ae-45df-b76d-134629d95281.png)

But we are usually using it to just connect commit / PR name with the version to know which one to upload:

![image](https://user-images.githubusercontent.com/6011991/207019597-eb1296f5-fccd-4b13-86f2-b76c4791aaa3.png)

**Comment by cincuranet** [comment]
Ahh, got it. I thought you already published it into our feed. I'll leave that for @LoopedBard3.

**Comment by anjiltech** [comment]
Post above changes (Updating the BDN version to 0.13.2.2043) within locally cloned environment and building the repo, it resolved the original issue related to RuntimeInformation and failing further with below messages:

------------------------------------------------------------------------
You should select the target benchmark(s). Please, print a number of a benchmark (e.g. `0`) or a contained benchmark caption (e.g. `BilinearTest`).
If you want to select few, please separate them with space ` ` (e.g. `1 2 3`).
You can also provide the class name in console arguments by using --filter. (e.g. `--filter '*BilinearTest*'`).
473

// Validating benchmarks:
// Benchmark Perf_Encoding.GetBytes: Job-OZWLNI(PowerPlanMode=00000000-0000-0000-0000-000000000000, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1) [size=16, encName=ascii]
//    * Mono is not installed or added to PATH, benchmark 'Perf_Encoding.GetBytes: Job-OZWLNI(PowerPlanMode=00000000-0000-0000-0000-000000000000, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1) [size=16, encName=ascii]' will not be executed

// Benchmark Perf_Encoding.GetBytes: Job-OZWLNI(PowerPlanMode=00000000-0000-0000-0000-000000000000, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1) [size=16, encName=ascii]
//    * The Roslyn toolchain is only supported on .NET Framework

// Benchmark Perf_Encoding.GetString: Job-OZWLNI(PowerPlanMode=00000000-0000-0000-0000-000000000000, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1) [size=16, encName=ascii]
//    * Mono is not installed or added to PATH, benchmark 'Perf_Encoding.GetString: Job-OZWLNI(PowerPlanMode=00000000-0000-0000-0000-000000000000, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1) [size=16, encName=ascii]' will not be executed

// Benchmark Perf_Encoding.GetString: Job-OZWLNI(PowerPlanMode=00000000-0000-0000-0000-000000000000, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1) [size=16, encName=ascii]
//    * The Roslyn toolchain is only supported on .NET Framework
---------------------------------------------------------------------------------

It is expecting Mono installation. While the dotnet info shows below runtimes and SDK installed:

$ dotnet --list-runtimes
Microsoft.AspNetCore.App 7.0.0 [/usr/lib64/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.NETCore.App 7.0.0 [/usr/lib64/dotnet/shared/Microsoft.NETCore.App]

$ dotnet --list-sdks
7.0.100 [/usr/lib64/dotnet/sdk]

**Comment by adamsitnik** [comment]
@anjiltech how do you run the benchmarks?

Could you please try running following command and share the output?

```cmd
dotnet run -c Release -f net7.0 --filter '*Perf_Encoding.GetBytes*'
```

**Comment by anjiltech** [comment]
@adamsitnik Please see below the command I was running and response
$ pwd
~/performance/src/benchmarks/micro

micro]$ dotnet run -c Release --framework net7.0 --filter '*Perf_Encoding.GetBytes*'
// Validating benchmarks:
// Benchmark Perf_Encoding.GetBytes: Job-YXYIDC(PowerPlanMode=00000000-0000-0000-0000-000000000000, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1) [size=16, encName=ascii]
//    * Mono is not installed or added to PATH, benchmark 'Perf_Encoding.GetBytes: Job-YXYIDC(PowerPlanMode=00000000-0000-0000-0000-000000000000, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1) [size=16, encName=ascii]' will not be executed

// Benchmark Perf_Encoding.GetBytes: Job-YXYIDC(PowerPlanMode=00000000-0000-0000-0000-000000000000, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1) [size=16, encName=ascii]
//    * The Roslyn toolchain is only supported on .NET Framework

// Benchmark Perf_Encoding.GetBytes: Job-YXYIDC(PowerPlanMode=00000000-0000-0000-0000-000000000000, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1) [size=16, encName=utf-8]
//    * Mono is not installed or added to PATH, benchmark 'Perf_Encoding.GetBytes: Job-YXYIDC(PowerPlanMode=00000000-0000-0000-0000-000000000000, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1) [size=16, encName=utf-8]' will not be executed

// Benchmark Perf_Encoding.GetBytes: Job-YXYIDC(PowerPlanMode=00000000-0000-0000-0000-000000000000, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1) [size=16, encName=utf-8]
//    * The Roslyn toolchain is only supported on .NET Framework

**Comment by adamsitnik** [comment]
I am unable to reproduce the issue you have mentioned on my Windows machine:

```log
BenchmarkDotNet=v0.13.2.2043-nightly, OS=Windows 11 (10.0.22621.819)
AMD Ryzen Threadripper PRO 3945WX 12-Cores, 1 CPU, 24 logical and 12 physical cores
.NET SDK=8.0.100-alpha.1.22558.1
  [Host]     : .NET 7.0.1 (7.0.122.56804), X64 RyuJIT AVX2
  Job-VUDPCD : .NET 7.0.1 (7.0.122.56804), X64 RyuJIT AVX2

PowerPlanMode=00000000-0000-0000-0000-000000000000  IterationTime=250.0000 ms  MaxIterationCount=20
MinIterationCount=15  WarmupCount=1

|   Method | size | encName |     Mean |    Error |   StdDev |   Median |      Min |      Max |   Gen0 | Allocated |
|--------- |----- |-------- |---------:|---------:|---------:|---------:|---------:|---------:|-------:|----------:|
| GetBytes |   16 |   ascii | 19.10 ns | 0.128 ns | 0.119 ns | 19.08 ns | 18.87 ns | 19.28 ns | 0.0047 |      40 B |
| GetBytes |   16 |   utf-8 | 17.14 ns | 0.065 ns | 0.054 ns | 17.14 ns | 17.02 ns | 17.20 ns | 0.0048 |      40 B |
| GetBytes |  512 |   ascii | 43.73 ns | 0.426 ns | 0.356 ns | 43.70 ns | 43.26 ns | 44.40 ns | 0.0640 |     536 B |
| GetBytes |  512 |   utf-8 | 63.85 ns | 0.367 ns | 0.343 ns | 63.74 ns | 63.36 ns | 64.51 ns | 0.0638 |     536 B |
```

I am going to try with Linux.

**Comment by adamsitnik** [comment]
@anjiltech I can't repro it on Linux:

```log
BenchmarkDotNet=v0.13.2.2043-nightly, OS=ubuntu 18.04
Intel Xeon CPU E5-1650 v4 3.60GHz, 1 CPU, 12 logical and 6 physical cores
.NET SDK=7.0.100-rc.1.22431.11
  [Host]     : .NET 7.0.0 (7.0.22.42316), X64 RyuJIT AVX2
  Job-LOEDTK : .NET 7.0.0 (7.0.22.42316), X64 RyuJIT AVX2

PowerPlanMode=00000000-0000-0000-0000-000000000000  IterationTime=250.0000 ms  MaxIterationCount=20  
MinIterationCount=15  WarmupCount=1  

|   Method | size | encName |     Mean |    Error |   StdDev |   Median |      Min |       Max |   Gen0 | Allocated |
|--------- |----- |-------- |---------:|---------:|---------:|---------:|---------:|----------:|-------:|----------:|
| GetBytes |   16 |   ascii | 21.74 ns | 0.255 ns | 0.226 ns | 21.72 ns | 21.50 ns |  22.45 ns | 0.0050 |      40 B |
| GetBytes |   16 |   utf-8 | 21.99 ns | 0.088 ns | 0.078 ns | 21.99 ns | 21.88 ns |  22.10 ns | 0.0050 |      40 B |
| GetBytes |  512 |   ascii | 72.95 ns | 0.484 ns | 0.429 ns | 72.84 ns | 72.51 ns |  74.05 ns | 0.0683 |     536 B |
| GetBytes |  512 |   utf-8 | 99.70 ns | 1.058 ns | 0.990 ns | 99.64 ns | 98.29 ns | 101.18 ns | 0.0680 |     536 B |
```

**Comment by anjiltech** [comment]
Hi @adamsitnik, the issue is reproducible on the pcc64le Linux environment. We tried on x86 Linux (RHEL) and it didnt reproduce.

I guess issue is specific to the newly added ppc64le architecture support.

**Comment by adamsitnik** [comment]
@anjiltech do you have some local changes? Do you use some kind of custom .NET SDK?

If not, could you please clone BenchmarkDotNet repo and try to run its test suite?

```cmd
git clone https://github.com/dotnet/BenchmarkDotNet.git repro
cd repro
./build.sh
```

**Comment by anjiltech** [comment]
[root@~ repro]# ./build.sh
dotnet-install: Note that the intended use of this script is for Continuous Integration (CI) scenarios, where:
dotnet-install: - The SDK needs to be installed without user interaction and without admin rights.
dotnet-install: - The SDK installation doesn't need to persist across multiple CI runs.
dotnet-install: To set up a development environment or to run apps, use installers rather than this script. Visit https://dotnet.microsoft.com/download to get the installer.

dotnet-install: .NET Core SDK with version '7.0.100' is already installed.
./build.sh: line 33: ~/repro/.dotnet/dotnet: cannot execute binary file: Exec format error
-------------------------------------------------------------------------------------------------------------
Looks like the 'dotnet' executable copied into  .dotnet/ is of x86 binary.  

We had below .NET SDK installed through standard RHEL repository.

[root@~]# sudo yum install dotnet-sdk-7.0 -y

Red Hat CodeReady Linux Builder for RHEL 8 Power, little endian (RPMs)                                                       
Package dotnet-sdk-7.0-7.0.100-1.el8_7.ppc64le is already installed.
Dependencies resolved.
Nothing to do.
Complete!

**Comment by anjiltech** [comment]
I just initiated the 'dotnet run' command directly (Instead of running build.sh , which seem to refer to the packaged 'dotnet' executable) and it seemed to build fine (Its been running for some time) 

dotnet run --project ./build/Build.csproj -- "$@"

....
....
Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:03:12.15

**Comment by anjiltech** [comment]
But the Test execution seem to fail:

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
System.TypeInitializationException: The type initializer for 'Microsoft.VisualStudio.TestPlatform.ObjectModel.Constants' threw an exception.
 ---> System.NotSupportedException: Specified method is not supported.
   at Microsoft.VisualStudio.TestPlatform.PlatformAbstractions.PlatformEnvironment.get_Architecture()
   at Microsoft.VisualStudio.TestPlatform.ObjectModel.Utilities.XmlRunSettingsUtilities.get_OSArchitecture()
   at Microsoft.VisualStudio.TestPlatform.ObjectModel.Constants..cctor()
   --- End of inner exception stack trace ---
   at Microsoft.VisualStudio.TestPlatform.ObjectModel.RunConfiguration.FromXml(XmlReader reader)
   at Microsoft.VisualStudio.TestPlatform.ObjectModel.Utilities.XmlRunSettingsUtilities.GetNodeValue[RunConfiguration](String settingsXml, String nodeName, Func`2 nodeParser)
   at Microsoft.VisualStudio.TestPlatform.ObjectModel.Utilities.XmlRunSettingsUtilities.GetRunConfigurationNode(String settingsXml)
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Execution.BaseRunTests.SetContext()
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Execution.BaseRunTests..ctor(IRequestData requestData, String package, String runSettings, TestExecutionContext testExecutionContext, ITestCaseEventsHandler testCaseEventsHandler, ITestRunEventsHandler testRunEventsHandler, ITestPlatformEventSource testPlatformEventSource, ITestEventsPublisher testEventsPublisher, IThread platformThread, IDataSerializer dataSerializer)
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Execution.BaseRunTests..ctor(IRequestData requestData, String package, String runSettings, TestExecutionContext testExecutionContext, ITestCaseEventsHandler testCaseEventsHandler, ITestRunEventsHandler testRunEventsHandler, ITestPlatformEventSource testPlatformEventSource)
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Execution.RunTestsWithSources..ctor(IRequestData requestData, Dictionary`2 adapterSourceMap, String package, String runSettings, TestExecutionContext testExecutionContext, ITestCaseEventsHandler testCaseEventsHandler, ITestRunEventsHandler testRunEventsHandler, Dictionary`2 executorUriVsSourceList)
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Execution.RunTestsWithSources..ctor(IRequestData requestData, Dictionary`2 adapterSourceMap, String package, String runSettings, TestExecutionContext testExecutionContext, ITestCaseEventsHandler testCaseEventsHandler, ITestRunEventsHandler testRunEventsHandler)
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Execution.ExecutionManager.StartTestRun(Dictionary`2 adapterSourceMap, String package, String runSettings, TestExecutionContext testExecutionContext, ITestCaseEventsHandler testCaseEventsHandler, ITestRunEventsHandler runEventsHandler)
Results File: ~/repro/TestResults/UnitTests-net7.0.trx

**Comment by adamsitnik** [comment]
@anjiltech in such a case please run following command from the repo root:

```cmd
dotnet test -c Release ./BenchmarkDotNet.sln
```

**Comment by anjiltech** [comment]
We see below format of exceptions with dotnet test:

A total of 1 test files matched the specified pattern.
  BenchmarkDotNet.IntegrationTests -> /home/anchinna/repro/tests/BenchmarkDotNet.IntegrationTests/bin/Release/net7.0/BenchmarkDotNet.IntegrationTests.dll
System.IO.FileNotFoundException: Could not find 'mono' host. Make sure that 'mono' is installed on the machine and is available in PATH environment variable.
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Helpers.DotnetHostHelper.GetMonoPath()
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Hosting.DefaultTestHostManager.GetTestHostProcessStartInfo(IEnumerable`1 sources, IDictionary`2 environmentVariables, TestRunnerConnectionInfo connectionInfo)
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyOperationManager.SetupChannel(IEnumerable`1 sources, String runSettings)
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyExecutionManager.StartTestRun(TestRunCriteria testRunCriteria, IInternalTestRunEventsHandler eventHandler)

Test Run Aborted.
Test run for /home/anchinna/repro/tests/BenchmarkDotNet.IntegrationTests.ManualRunning/bin/Release/net7.0/BenchmarkDotNet.IntegrationTests.ManualRunning.dll (.NETCoreApp,Version=v7.0)
Microsoft (R) Test Execution Command Line Tool Version 17.4.0+c02ece877c62577810f893c44279ce79af820112 (ppc64le)
Copyright (c) Microsoft Corporation.  All rights reserved.

System.TypeInitializationException: The type initializer for 'Microsoft.VisualStudio.TestPlatform.ObjectModel.Constants' threw an exception.
 ---> System.NotSupportedException: Specified method is not supported.
   at Microsoft.VisualStudio.TestPlatform.PlatformAbstractions.PlatformEnvironment.get_Architecture()
   at Microsoft.VisualStudio.TestPlatform.ObjectModel.Utilities.XmlRunSettingsUtilities.get_OSArchitecture()
   at Microsoft.VisualStudio.TestPlatform.ObjectModel.Constants..cctor()
   --- End of inner exception stack trace ---
   at Microsoft.VisualStudio.TestPlatform.ObjectModel.RunConfiguration.FromXml(XmlReader reader)
   at Microsoft.VisualStudio.TestPlatform.ObjectModel.Utilities.XmlRunSettingsUtilities.GetNodeValue[RunConfiguration](String settingsXml, String nodeName, Func`2 nodeParser)
   at Microsoft.VisualStudio.TestPlatform.ObjectModel.Utilities.XmlRunSettingsUtilities.GetRunConfigurationNode(String settingsXml)
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Execution.BaseRunTests.SetContext()
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Execution.BaseRunTests..ctor(IRequestData requestData, String package, String runSettings, TestExecutionContext testExecutionContext, ITestCaseEventsHandler testCaseEventsHandler, ITestRunEventsHandler testRunEventsHandler, ITestPlatformEventSource testPlatformEventSource, ITestEventsPublisher testEventsPublisher, IThread platformThread, IDataSerializer dataSerializer)
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Execution.BaseRunTests..ctor(IRequestData requestData, String package, String runSettings, TestExecutionContext testExecutionContext, ITestCaseEventsHandler testCaseEventsHandler, ITestRunEventsHandler testRunEventsHandler, ITestPlatformEventSource testPlatformEventSource)
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Execution.RunTestsWithSources..ctor(IRequestData requestData, Dictionary`2 adapterSourceMap, String package, String runSettings, TestExecutionContext testExecutionContext, ITestCaseEventsHandler testCaseEventsHandler, ITestRunEventsHandler testRunEventsHandler, Dictionary`2 executorUriVsSourceList)
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Execution.RunTestsWithSources..ctor(IRequestData requestData, Dictionary`2 adapterSourceMap, String package, String runSettings, TestExecutionContext testExecutionContext, ITestCaseEventsHandler testCaseEventsHandler, ITestRunEventsHandler testRunEventsHandler)
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Execution.ExecutionManager.StartTestRun(Dictionary`2 adapterSourceMap, String package, String runSettings, TestExecutionContext testExecutionContext, ITestCaseEventsHandler testCaseEventsHandler, ITestRunEventsHandler runEventsHandler)
Starting test execution, please wait...

Test Run Aborted.

**Comment by adamsitnik** [comment]
This is most likely lack of ppc64le support for `dotnet test`:

```log
at Microsoft.VisualStudio.TestPlatform.PlatformAbstractions.PlatformEnvironment.get_Architecture()
```

But this seems to be an invalid runtime recognition:

```log
System.IO.FileNotFoundException: Could not find 'mono' host. Make sure that 'mono' is installed on the machine and is available in PATH environment variable.
at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Helpers.DotnetHostHelper.GetMonoPath()
```

It looks like your SDK does not detect the current runtime as CLR, but as Mono. Based on the previous logs you have shared I believe that BenchmarkDotNet does the same. @anjiltech But perhaps it does it on purpose? I know that Mono supports more architectures than CLR. Who owns the SDK that you are using?

Could you please see what `IsMono` of [RuntimeInformation](https://github.com/dotnet/BenchmarkDotNet/blob/69928565a98a26d489611a966d99e90ce52f1952/src/BenchmarkDotNet/Portability/RuntimeInformation.cs#L22) class return? The simplest way to do it would be to make this class public and just print it from `Program.cs` in https://github.com/dotnet/BenchmarkDotNet/blob/master/samples/BenchmarkDotNet.Samples/Program.cs#L7

**Comment by anjiltech** [comment]
Hi @adamsitnik , I got to learn that the current version of the .NET SDK for ppc64le architecture only supports Mono runtime and not the CLR.  

With this behaviour of the SDK, do we see any issues with `BenchmarkDotNet` and `performance`  repos to work with Mono?

**Comment by adamsitnik** [comment]
> With this behaviour of the SDK, do we see any issues with `BenchmarkDotNet` and `performance`  repos to work with Mono? 

I do at least one (BDN recognizing the new Mono as the old Mono), but I've sent a fix for it:

https://github.com/dotnet/BenchmarkDotNet/pull/2230

@anjiltech  Could you please clone this branch and see if it solves the problem?

```cmd
git clone https://github.com/dotnet/BenchmarkDotNet.git --branch monoSDK monoSDK 
cd monoSDK/samples/BenchmarkDotNet.Samples/
dotnet run -c Release -f net7.0 --filter BenchmarkDotNet.Samples.IntroBasic.Sleep --job dry
```

The only other place that might not work is https://github.com/dotnet/BenchmarkDotNet/blob/69928565a98a26d489611a966d99e90ce52f1952/src/BenchmarkDotNet/Toolchains/Mono/MonoPublisher.cs#L18 but I need your help with ppc64le testing.

**Comment by anjiltech** [comment]
@adamsitnik  With above clone (monoSDK), it fails with below exception

BenchmarkDotNet.Samples]$ dotnet run -c Release -f net7.0 --filter BenchmarkDotNet.Samples.IntroBasic.Sleep --job dry

Unhandled Exception:
System.NotSupportedException: Ppc64le is not supported
   at BenchmarkDotNet.Disassemblers.SameArchitectureDisassembler.CreateDisassemblerForCurrentArchitecture() in /home/anchinna/monoSDK/src/BenchmarkDotNet/Disassemblers/SameArchitectureDisassembler.cs:line 28
   at BenchmarkDotNet.Disassemblers.SameArchitectureDisassembler..ctor(DisassemblyDiagnoserConfig config) in /home/anchinna/monoSDK/src/BenchmarkDotNet/Disassemblers/SameArchitectureDisassembler.cs:line 17
   at BenchmarkDotNet.Diagnosers.DisassemblyDiagnoser..ctor(DisassemblyDiagnoserConfig config) in /home/anchinna/monoSDK/src/BenchmarkDotNet/Disassemblers/DisassemblyDiagnoser.cs:line 35
   at BenchmarkDotNet.Attributes.DisassemblyDiagnoserAttribute..ctor(Int32 maxDepth, DisassemblySyntax syntax, Boolean printSource, Boolean printInstructionAddresses, Boolean exportGithubMarkdown, Boolean exportHtml, Boolean exportCombinedDisassemblyReport, Boolean exportDiff, String[] filters) in /home/anchinna/monoSDK/src/BenchmarkDotNet/Attributes/DisassemblyDiagnoserAttribute.cs:line 30
   at System.Reflection.CustomAttribute.GetCustomAttributesBase(ICustomAttributeProvider obj, Type attributeType, Boolean inheritedOnly)
   at System.Reflection.CustomAttribute.GetCustomAttributes(ICustomAttributeProvider obj, Type attributeType, Boolean inherit)
   at System.RuntimeType.GetCustomAttributes(Boolean inherit)
   at BenchmarkDotNet.Helpers.GenericBenchmarksBuilder.BuildGenericsIfNeeded(Type type) in /home/anchinna/monoSDK/src/BenchmarkDotNet/Helpers/GenericBenchmarksBuilder.cs:line 20

**Comment by adamsitnik** [comment]
The disassembler samples are causing the exception, you can just remove them:

```cmd
cd monoSDK/samples/BenchmarkDotNet.Samples/
rm ./IntroDisassembly*
```

**Comment by anjiltech** [comment]
@adamsitnik , I deleted those  IntroDisassembly* files and we hit same issue, can you please see the stack trace below to see the flow thats triggering Disassemblers.

BenchmarkDotNet.Samples]$ dotnet run -c Release -f net7.0 --filter BenchmarkDotNet.Samples.IntroBasic.Sleep --job dry

Unhandled Exception:
System.NotSupportedException: Ppc64le is not supported
   at BenchmarkDotNet.Disassemblers.SameArchitectureDisassembler.CreateDisassemblerForCurrentArchitecture() in /home/anchinna/monoSDK/src/BenchmarkDotNet/Disassemblers/SameArchitectureDisassembler.cs:line 28
   at BenchmarkDotNet.Disassemblers.SameArchitectureDisassembler..ctor(DisassemblyDiagnoserConfig config) in /home/anchinna/monoSDK/src/BenchmarkDotNet/Disassemblers/SameArchitectureDisassembler.cs:line 17
   at BenchmarkDotNet.Diagnosers.DisassemblyDiagnoser..ctor(DisassemblyDiagnoserConfig config) in /home/anchinna/monoSDK/src/BenchmarkDotNet/Disassemblers/DisassemblyDiagnoser.cs:line 35
   at BenchmarkDotNet.Diagnosers.DiagnosersLoader.LoadDiagnosers()+MoveNext() in /home/anchinna/monoSDK/src/BenchmarkDotNet/Diagnosers/DiagnosersLoader.cs:line 37
   at System.Collections.Generic.LargeArrayBuilder`1[[BenchmarkDotNet.Diagnosers.IDiagnoser, BenchmarkDotNet, Version=0.13.2.0, Culture=neutral, PublicKeyToken=aa0ca2f9092cefc4]].AddRange(IEnumerable`1 items)
   at System.Collections.Generic.EnumerableHelpers.ToArray[IDiagnoser](IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[IDiagnoser](IEnumerable`1 source)
   at BenchmarkDotNet.Diagnosers.DiagnosersLoader.<>c.<.cctor>b__9_0() in /home/anchinna/monoSDK/src/BenchmarkDotNet/Diagnosers/DiagnosersLoader.cs:line 20
   at System.Lazy`1[[BenchmarkDotNet.Diagnosers.IDiagnoser[], BenchmarkDotNet, Version=0.13.2.0, Culture=neutral, PublicKeyToken=aa0ca2f9092cefc4]].ViaFactory(LazyThreadSafetyMode mode)
   at System.Lazy`1[[BenchmarkDotNet.Diagnosers.IDiagnoser[], BenchmarkDotNet, Version=0.13.2.0, Culture=neutral, PublicKeyToken=aa0ca2f9092cefc4]].ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
   at System.Lazy`1[[BenchmarkDotNet.Diagnosers.IDiagnoser[], BenchmarkDotNet, Version=0.13.2.0, Culture=neutral, PublicKeyToken=aa0ca2f9092cefc4]].CreateValue()
   at System.Lazy`1[[BenchmarkDotNet.Diagnosers.IDiagnoser[], BenchmarkDotNet, Version=0.13.2.0, Culture=neutral, PublicKeyToken=aa0ca2f9092cefc4]].get_Value()
   at BenchmarkDotNet.Diagnosers.DiagnosersLoader.GetImplementation[IHardwareCountersDiagnoser]() in /home/anchinna/monoSDK/src/BenchmarkDotNet/Diagnosers/DiagnosersLoader.cs:line 23
   at BenchmarkDotNet.Configs.ImmutableConfigBuilder.GetDiagnosers(IEnumerable`1 diagnosers, ImmutableHashSet`1 uniqueHardwareCounters) in /home/anchinna/monoSDK/src/BenchmarkDotNet/Configs/ImmutableConfigBuilder.cs:line 89
   at BenchmarkDotNet.Configs.ImmutableConfigBuilder.Create(IConfig source) in /home/anchinna/monoSDK/src/BenchmarkDotNet/Configs/ImmutableConfigBuilder.cs:line 43
   at BenchmarkDotNet.Running.BenchmarkConverter.GetFullTypeConfig(Type type, IConfig config) in /home/anchinna/monoSDK/src/BenchmarkDotNet/Running/BenchmarkConverter.cs:line 95
   at BenchmarkDotNet.Running.BenchmarkConverter.MethodsToBenchmarksWithFullConfig(Type type, MethodInfo[] benchmarkMethods, IConfig config) in /home/anchinna/monoSDK/src/BenchmarkDotNet/Running/BenchmarkConverter.cs:line 47
   at BenchmarkDotNet.Running.BenchmarkConverter.TypeToBenchmarks(Type type, IConfig config) in /home/anchinna/monoSDK/src/BenchmarkDotNet/Running/BenchmarkConverter.cs:line 29
   at BenchmarkDotNet.Running.TypeFilter.<>c__DisplayClass1_0.<Filter>b__0(Type type) in /home/anchinna/monoSDK/src/BenchmarkDotNet/Running/TypeFilter.cs:line 42
   at System.Linq.Enumerable.SelectListIterator`2[[System.Type, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[BenchmarkDotNet.Running.BenchmarkRunInfo, BenchmarkDotNet, Version=0.13.2.0, Culture=neutral, PublicKeyToken=aa0ca2f9092cefc4]].MoveNext()
   at System.Linq.Enumerable.WhereEnumerableIterator`1[[BenchmarkDotNet.Running.BenchmarkRunInfo, BenchmarkDotNet, Version=0.13.2.0, Culture=neutral, PublicKeyToken=aa0ca2f9092cefc4]].ToArray()
   at System.Linq.Enumerable.ToArray[BenchmarkRunInfo](IEnumerable`1 source)
   at BenchmarkDotNet.Running.TypeFilter.Filter(IConfig effectiveConfig, IEnumerable`1 types) in /home/anchinna/monoSDK/src/BenchmarkDotNet/Running/TypeFilter.cs:line 41
   at BenchmarkDotNet.Running.BenchmarkSwitcher.RunWithDirtyAssemblyResolveHelper(String[] args, IConfig config, Boolean askUserForInput) in /home/anchinna/monoSDK/src/BenchmarkDotNet/Running/BenchmarkSwitcher.cs:line 120
   at BenchmarkDotNet.Running.BenchmarkSwitcher.Run(String[] args, IConfig config) in /home/anchinna/monoSDK/src/BenchmarkDotNet/Running/BenchmarkSwitcher.cs:line 70
   at BenchmarkDotNet.Samples.Program.Main(String[] args) in /home/anchinna/monoSDK/samples/BenchmarkDotNet.Samples/Program.cs:line 7
[ERROR] FATAL UNHANDLED EXCEPTION: System.NotSupportedException: Ppc64le is not supported
   at BenchmarkDotNet.Disassemblers.SameArchitectureDisassembler.CreateDisassemblerForCurrentArchitecture() in /home/anchinna/monoSDK/src/BenchmarkDotNet/Disassemblers/SameArchitectureDisassembler.cs:line 28
   at BenchmarkDotNet.Disassemblers.SameArchitectureDisassembler..ctor(DisassemblyDiagnoserConfig config) in /home/anchinna/monoSDK/src/BenchmarkDotNet/Disassemblers/SameArchitectureDisassembler.cs:line 17
   at BenchmarkDotNet.Diagnosers.DisassemblyDiagnoser..ctor(DisassemblyDiagnoserConfig config) in /home/anchinna/monoSDK/src/BenchmarkDotNet/Disassemblers/DisassemblyDiagnoser.cs:line 35
   at BenchmarkDotNet.Diagnosers.DiagnosersLoader.LoadDiagnosers()+MoveNext() in /home/anchinna/monoSDK/src/BenchmarkDotNet/Diagnosers/DiagnosersLoader.cs:line 37
   at System.Collections.Generic.LargeArrayBuilder`1[[BenchmarkDotNet.Diagnosers.IDiagnoser, BenchmarkDotNet, Version=0.13.2.0, Culture=neutral, PublicKeyToken=aa0ca2f9092cefc4]].AddRange(IEnumerable`1 items)
   at System.Collections.Generic.EnumerableHelpers.ToArray[IDiagnoser](IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[IDiagnoser](IEnumerable`1 source)
   at BenchmarkDotNet.Diagnosers.DiagnosersLoader.<>c.<.cctor>b__9_0() in /home/anchinna/monoSDK/src/BenchmarkDotNet/Diagnosers/DiagnosersLoader.cs:line 20
   at System.Lazy`1[[BenchmarkDotNet.Diagnosers.IDiagnoser[], BenchmarkDotNet, Version=0.13.2.0, Culture=neutral, PublicKeyToken=aa0ca2f9092cefc4]].ViaFactory(LazyThreadSafetyMode mode)
   at System.Lazy`1[[BenchmarkDotNet.Diagnosers.IDiagnoser[], BenchmarkDotNet, Version=0.13.2.0, Culture=neutral, PublicKeyToken=aa0ca2f9092cefc4]].ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
   at System.Lazy`1[[BenchmarkDotNet.Diagnosers.IDiagnoser[], BenchmarkDotNet, Version=0.13.2.0, Culture=neutral, PublicKeyToken=aa0ca2f9092cefc4]].CreateValue()
   at System.Lazy`1[[BenchmarkDotNet.Diagnosers.IDiagnoser[], BenchmarkDotNet, Version=0.13.2.0, Culture=neutral, PublicKeyToken=aa0ca2f9092cefc4]].get_Value()
   at BenchmarkDotNet.Diagnosers.DiagnosersLoader.GetImplementation[IHardwareCountersDiagnoser]() in /home/anchinna/monoSDK/src/BenchmarkDotNet/Diagnosers/DiagnosersLoader.cs:line 23
   at BenchmarkDotNet.Configs.ImmutableConfigBuilder.GetDiagnosers(IEnumerable`1 diagnosers, ImmutableHashSet`1 uniqueHardwareCounters) in /home/anchinna/monoSDK/src/BenchmarkDotNet/Configs/ImmutableConfigBuilder.cs:line 89
   at BenchmarkDotNet.Configs.ImmutableConfigBuilder.Create(IConfig source) in /home/anchinna/monoSDK/src/BenchmarkDotNet/Configs/ImmutableConfigBuilder.cs:line 43
   at BenchmarkDotNet.Running.BenchmarkConverter.GetFullTypeConfig(Type type, IConfig config) in /home/anchinna/monoSDK/src/BenchmarkDotNet/Running/BenchmarkConverter.cs:line 95
   at BenchmarkDotNet.Running.BenchmarkConverter.MethodsToBenchmarksWithFullConfig(Type type, MethodInfo[] benchmarkMethods, IConfig config) in /home/anchinna/monoSDK/src/BenchmarkDotNet/Running/BenchmarkConverter.cs:line 47
   at BenchmarkDotNet.Running.BenchmarkConverter.TypeToBenchmarks(Type type, IConfig config) in /home/anchinna/monoSDK/src/BenchmarkDotNet/Running/BenchmarkConverter.cs:line 29
   at BenchmarkDotNet.Running.TypeFilter.<>c__DisplayClass1_0.<Filter>b__0(Type type) in /home/anchinna/monoSDK/src/BenchmarkDotNet/Running/TypeFilter.cs:line 42
   at System.Linq.Enumerable.SelectListIterator`2[[System.Type, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[BenchmarkDotNet.Running.BenchmarkRunInfo, BenchmarkDotNet, Version=0.13.2.0, Culture=neutral, PublicKeyToken=aa0ca2f9092cefc4]].MoveNext()
   at System.Linq.Enumerable.WhereEnumerableIterator`1[[BenchmarkDotNet.Running.BenchmarkRunInfo, BenchmarkDotNet, Version=0.13.2.0, Culture=neutral, PublicKeyToken=aa0ca2f9092cefc4]].ToArray()
   at System.Linq.Enumerable.ToArray[BenchmarkRunInfo](IEnumerable`1 source)
   at BenchmarkDotNet.Running.TypeFilter.Filter(IConfig effectiveConfig, IEnumerable`1 types) in /home/anchinna/monoSDK/src/BenchmarkDotNet/Running/TypeFilter.cs:line 41
   at BenchmarkDotNet.Running.BenchmarkSwitcher.RunWithDirtyAssemblyResolveHelper(String[] args, IConfig config, Boolean askUserForInput) in /home/anchinna/monoSDK/src/BenchmarkDotNet/Running/BenchmarkSwitcher.cs:line 120
   at BenchmarkDotNet.Running.BenchmarkSwitcher.Run(String[] args, IConfig config) in /home/anchinna/monoSDK/src/BenchmarkDotNet/Running/BenchmarkSwitcher.cs:line 70
   at BenchmarkDotNet.Samples.Program.Main(String[] args) in /home/anchinna/monoSDK/samples/BenchmarkDotNet.Samples/Program.cs:line 7

[ BenchmarkDotNet.Samples]$

**Comment by adamsitnik** [comment]
@anjiltech please fetch latest changes and try again

**Comment by anjiltech** [comment]
hi @adamsitnik , It fixed previous error and fails with a different issue:  "error NU1101: Unable to find package Microsoft.NETCore.App.Runtime.Mono.linux-ppc64le" 

Can you please see below trace:

BenchmarkDotNet.Samples]$ dotnet run -c Release -f net7.0 --filter BenchmarkDotNet.Samples.IntroBasic.Sleep --job dry
// Validating benchmarks:
// Wrong assembly binding redirects for System.Drawing.Common, loading it from disk anyway.
// ***** BenchmarkRunner: Start   *****
// ***** Found 1 benchmark(s) in total *****
// ***** Building 1 exe(s) in Parallel: Start   *****
// start dotnet  publish -c Release --self-contained -r linux-ppc64le /p:UseMonoRuntime=true /p:RuntimeIdentifiers=linux-ppc64le /p:UseSharedCompilation=false /p:BuildInParallel=false /m:1 /p:Deterministic=true /p:Optimize=true --output "/home/anchinna/monoSDK/samples/BenchmarkDotNet.Samples/bin/Release/net7.0/ef322eff-0dc7-4163-abe7-c162c07c51a1/bin/Release/net7.0" in /home/anchinna/monoSDK/samples/BenchmarkDotNet.Samples/bin/Release/net7.0/ef322eff-0dc7-4163-abe7-c162c07c51a1
// command took 5.4s and exited with 1
// ***** Done, took 00:00:05 (5.62 sec)   *****
// Found 1 benchmarks:
//   IntroBasic.Sleep: Dry(IterationCount=1, LaunchCount=1, RunStrategy=ColdStart, UnrollFactor=1, WarmupCount=1)

// Build Error: Standard output:

 Standard error:
 MSBuild version 17.4.0+18d5aef85 for .NET
  Determining projects to restore...
  Restored /home/anchinna/monoSDK/src/BenchmarkDotNet.Annotations/BenchmarkDotNet.Annotations.csproj (in 663 ms).
  Restored /home/anchinna/monoSDK/src/BenchmarkDotNet.Diagnostics.Windows/BenchmarkDotNet.Diagnostics.Windows.csproj (in 663 ms).
/home/anchinna/monoSDK/src/BenchmarkDotNet/BenchmarkDotNet.csproj : error NU1101: Unable to find package Microsoft.NETCore.App.Runtime.Mono.linux-ppc64le. No packages exist with this id in source(s): api.nuget.org, dotnet6, dotnet7, dotnet8 [/home/anchinna/monoSDK/samples/BenchmarkDotNet.Samples/bin/Release/net7.0/ef322eff-0dc7-4163-abe7-c162c07c51a1/BenchmarkDotNet.Autogenerated.csproj]
/home/anchinna/monoSDK/samples/BenchmarkDotNet.Samples/bin/Release/net7.0/ef322eff-0dc7-4163-abe7-c162c07c51a1/BenchmarkDotNet.Autogenerated.csproj : error NU1101: Unable to find package Microsoft.NETCore.App.Runtime.Mono.linux-ppc64le. No packages exist with this id in source(s): api.nuget.org, dotnet6, dotnet7, dotnet8
/home/anchinna/monoSDK/samples/BenchmarkDotNet.Samples/bin/Release/net7.0/ef322eff-0dc7-4163-abe7-c162c07c51a1/BenchmarkDotNet.Autogenerated.csproj : error NU1101: Unable to find package Microsoft.AspNetCore.App.Runtime.linux-ppc64le. No packages exist with this id in source(s): api.nuget.org, dotnet6, dotnet7, dotnet8
/home/anchinna/monoSDK/samples/BenchmarkDotNet.Samples/bin/Release/net7.0/ef322eff-0dc7-4163-abe7-c162c07c51a1/BenchmarkDotNet.Autogenerated.csproj : error NU1101: Unable to find package Microsoft.NETCore.App.Host.linux-ppc64le. No packages exist with this id in source(s): api.nuget.org, dotnet6, dotnet7, dotnet8
/home/anchinna/monoSDK/samples/BenchmarkDotNet.Samples/BenchmarkDotNet.Samples.csproj : error NU1101: Unable to find package Microsoft.NETCore.App.Runtime.Mono.linux-ppc64le. No packages exist with this id in source(s): api.nuget.org, dotnet6, dotnet7, dotnet8 [/home/anchinna/monoSDK/samples/BenchmarkDotNet.Samples/bin/Release/net7.0/ef322eff-0dc7-4163-abe7-c162c07c51a1/BenchmarkDotNet.Autogenerated.csproj]
/home/anchinna/monoSDK/samples/BenchmarkDotNet.Samples/BenchmarkDotNet.Samples.csproj : error NU1101: Unable to find package Microsoft.AspNetCore.App.Runtime.linux-ppc64le. No packages exist with this id in source(s): api.nuget.org, dotnet6, dotnet7, dotnet8 [/home/anchinna/monoSDK/samples/BenchmarkDotNet.Samples/bin/Release/net7.0/ef322eff-0dc7-4163-abe7-c162c07c51a1/BenchmarkDotNet.Autogenerated.csproj]
/home/anchinna/monoSDK/samples/BenchmarkDotNet.Samples/BenchmarkDotNet.Samples.csproj : error NU1101: Unable to find package Microsoft.NETCore.App.Host.linux-ppc64le. No packages exist with this id in source(s): api.nuget.org, dotnet6, dotnet7, dotnet8 [/home/anchinna/monoSDK/samples/BenchmarkDotNet.Samples/bin/Release/net7.0/ef322eff-0dc7-4163-abe7-c162c07c51a1/BenchmarkDotNet.Autogenerated.csproj]
  Failed to restore /home/anchinna/monoSDK/samples/BenchmarkDotNet.Samples/bin/Release/net7.0/ef322eff-0dc7-4163-abe7-c162c07c51a1/BenchmarkDotNet.Autogenerated.csproj (in 1.3 sec).
  Failed to restore /home/anchinna/monoSDK/src/BenchmarkDotNet/BenchmarkDotNet.csproj (in 1.35 sec).
  Failed to restore /home/anchinna/monoSDK/samples/BenchmarkDotNet.Samples/BenchmarkDotNet.Samples.csproj (in 1.38 sec).

// BenchmarkDotNet has failed to build the auto-generated boilerplate code.
// It can be found in /home/anchinna/monoSDK/samples/BenchmarkDotNet.Samples/bin/Release/net7.0/ef322eff-0dc7-4163-abe7-c162c07c51a1
// Please follow the troubleshooting guide: https://benchmarkdotnet.org/articles/guides/troubleshooting.html

**Comment by adamsitnik** [comment]
Hmm, we can't publish a self-contained Mono app. I wonder if there is any Mono specific NuGet feed?

@anjiltech  In the meantime could you please try following command (it will use different toolchain for building the app)

```cmd
dotnet run -c Release -f net7.0 --filter BenchmarkDotNet.Samples.IntroBasic.Sleep --job dry --runtimes net7.0
```

**Comment by adamsitnik** [comment]
@shivanirmishra I've pushed one more fix, please fetch it and give it a try:

```cmd
dotnet run -c Release -f net7.0 --filter BenchmarkDotNet.Samples.IntroBasic.Sleep --job dry
```

**Comment by shivanirmishra** [comment]
@adamsitnik Is this functionally fine?

BenchmarkDotNet.Samples]$ dotnet run -c Release -f net7.0 --filter BenchmarkDotNet.Samples.IntroBasic.Sleep --job dry
// Validating benchmarks:
// Wrong assembly binding redirects for System.Drawing.Common, loading it from disk anyway.
// ***** BenchmarkRunner: Start   *****
// ***** Found 1 benchmark(s) in total *****
// ***** Building 1 exe(s) in Parallel: Start   *****
// start dotnet  restore /p:UseSharedCompilation=false /p:BuildInParallel=false /m:1 /p:Deterministic=true /p:Optimize=true in /home/shivani/dotNet/monoSDK/samples/BenchmarkDotNet.Samples/bin/Release/net7.0/3b65c94c-d9b6-4e68-9687-fd29a09e4a3a
// command took 4.53s and exited with 0
// start dotnet  build -c Release --no-restore /p:UseSharedCompilation=false /p:BuildInParallel=false /m:1 /p:Deterministic=true /p:Optimize=true in /home/shivani/dotNet/monoSDK/samples/BenchmarkDotNet.Samples/bin/Release/net7.0/3b65c94c-d9b6-4e68-9687-fd29a09e4a3a
// command took 10.05s and exited with 0
// ***** Done, took 00:00:14 (14.79 sec)   *****
// Found 1 benchmarks:
//   IntroBasic.Sleep: Dry(IterationCount=1, LaunchCount=1, RunStrategy=ColdStart, UnrollFactor=1, WarmupCount=1)

// **************************
// Benchmark: IntroBasic.Sleep: Dry(IterationCount=1, LaunchCount=1, RunStrategy=ColdStart, UnrollFactor=1, WarmupCount=1)
// *** Execute ***
// Launch: 1 / 1
// Execute: dotnet 3b65c94c-d9b6-4e68-9687-fd29a09e4a3a.dll --anonymousPipes 18 19 --benchmarkName BenchmarkDotNet.Samples.IntroBasic.Sleep --job Dry --benchmarkId 0 in /home/shivani/dotNet/monoSDK/samples/BenchmarkDotNet.Samples/bin/Release/net7.0/3b65c94c-d9b6-4e68-9687-fd29a09e4a3a/bin/Release/net7.0
Failed to set up high priority. Make sure you have the right permissions. Message: Permission denied
// BeforeAnythingElse

// Benchmark Process Environment Information:
// Runtime=.NET 7.0.0 (7.0.22.52201) using MonoVM, Ppc64le 
// GC=Non-concurrent Workstation
// HardwareIntrinsics= 
// Job: Dry(IterationCount=1, LaunchCount=1, RunStrategy=ColdStart, UnrollFactor=1, WarmupCount=1)

// BeforeActualRun
WorkloadActual   1: 1 op, 12170830.00 ns, 12.1708 ms/op

// AfterActualRun
WorkloadResult   1: 1 op, 12170830.00 ns, 12.1708 ms/op

// AfterAll
// Benchmark Process 313422 has exited with code 0.

Mean = 12.171 ms, StdErr = 0.000 ms (0.00%), N = 1, StdDev = 0.000 ms
Min = 12.171 ms, Q1 = 12.171 ms, Median = 12.171 ms, Q3 = 12.171 ms, Max = 12.171 ms
IQR = 0.000 ms, LowerFence = 12.171 ms, UpperFence = 12.171 ms
ConfidenceInterval = [NaN ms; NaN ms] (CI 99.9%), Margin = NaN ms (NaN% of Mean)
Skewness = NaN, Kurtosis = NaN, MValue = 2

// ** Remained 0 (0.0%) benchmark(s) to run. Estimated finish 2022-12-15 5:04 (0h 0m from now) **
// ***** BenchmarkRunner: Finish  *****

// * Export *
  BenchmarkDotNet.Artifacts/results/BenchmarkDotNet.Samples.IntroBasic-report.csv
  BenchmarkDotNet.Artifacts/results/BenchmarkDotNet.Samples.IntroBasic-report-github.md
  BenchmarkDotNet.Artifacts/results/BenchmarkDotNet.Samples.IntroBasic-report.html

// * Detailed results *
IntroBasic.Sleep: Dry(IterationCount=1, LaunchCount=1, RunStrategy=ColdStart, UnrollFactor=1, WarmupCount=1)
Runtime = .NET 7.0.0 (7.0.22.52201) using MonoVM, Ppc64le ; GC = Non-concurrent Workstation
Mean = 12.171 ms, StdErr = 0.000 ms (0.00%), N = 1, StdDev = 0.000 ms
Min = 12.171 ms, Q1 = 12.171 ms, Median = 12.171 ms, Q3 = 12.171 ms, Max = 12.171 ms
IQR = 0.000 ms, LowerFence = 12.171 ms, UpperFence = 12.171 ms
ConfidenceInterval = [NaN ms; NaN ms] (CI 99.9%), Margin = NaN ms (NaN% of Mean)
Skewness = NaN, Kurtosis = NaN, MValue = 2
-------------------- Histogram --------------------
[12.171 ms ; 12.171 ms) | @
---------------------------------------------------

// * Summary *

BenchmarkDotNet=v0.13.2.20221215-develop, OS=rhel 8
Unknown processor
.NET SDK=7.0.100
  [Host] : .NET 7.0.0 (7.0.22.52201) using MonoVM, Ppc64le 
  Dry    : .NET 7.0.0 (7.0.22.52201) using MonoVM, Ppc64le 

Job=Dry  IterationCount=1  LaunchCount=1  
RunStrategy=ColdStart  UnrollFactor=1  WarmupCount=1  

| Method |     Mean | Error |
|------- |---------:|------:|
|  Sleep | 12.17 ms |    NA |

// * Warnings *
MinIterationTime
  IntroBasic.Sleep: Dry -> The minimum observed iteration time is 12.1708 ms which is very small. It's recommended to increase it to at least 100.0000 ms using more operations.

// * Legends *
  Mean  : Arithmetic mean of all measurements
  Error : Half of 99.9% confidence interval
  1 ms  : 1 Millisecond (0.001 sec)

// ***** BenchmarkRunner: End *****
Run time: 00:00:00 (0.5 sec), executed benchmarks: 1

Global total time: 00:00:15 (15.55 sec), executed benchmarks: 1
// * Artifacts cleanup *

**Comment by adamsitnik** [comment]
> Is this functionally fine?

Yes! The last few lines from following summary represent:
* `[Host]` - the host process information, which confirms that your SDK runs as MonoVM
* `Dry` - benchmark process information (in this case with Id=Dry because you have specified `--job dry` to execute the benchmark just once), which also confirms that benchmarks were executed as MonoVM  Ppc64le

```ini
BenchmarkDotNet=v0.13.2.20221215-develop, OS=rhel 8
Unknown processor
.NET SDK=7.0.100
[Host] : .NET 7.0.0 (7.0.22.52201) using MonoVM, Ppc64le
Dry : .NET 7.0.0 (7.0.22.52201) using MonoVM, Ppc64le
```

**Comment by shivanirmishra** [comment]
@adamsitnik is this getting merged into main BenchmarkDotNet branch and further into 'performance' repository?

**Comment by anjiltech** [comment]
Hi @LoopedBard3 Can the performance repo be updated with latest version of BDN [0.13.2.2052](https://ci.appveyor.com/project/dotnetfoundation/benchmarkdotnet/builds/45706682)
